overlord/devicestate/firstboot.go

```diff
@@ -317,6 +320,8 @@ func (m *DeviceManager) populateStateFromSeedImpl(tm timings.Measurer) ([]*state
 			// non-dangerous models, we can do that here since that information will
 			// probably be in the model assertion which we have here
 			ApplySnapDevMode: modelIsDangerous,
+			Lane:             essentialLane,
+			Transaction:      client.TransactionAllSnaps,
```

Comment:
what's the reasoning for the transaction change?

Reply:
this is immaterial though because we are creating a TaskSet for one snap only?

Reply:
what i mean is that check is there for multiple snaps, otherwise the two transaction modes are the same?  I understand you need to pass the value to make the check happy. I'm just wondering whether the requirement is confusing if there is only one snap in play